### ISSUE TITLE
publish does't merge remote git-annex upon publish, thus not "propagating" the knowledge about location of the load

### ISSUE THREAD

**Issue by yarikoptic** [datalad/datalad#765]
I thought it was working but apparently at least in this case somehow it doesn't:

```
(git)smaug:\u2026ts/datalad/crawl/kaggle/melbourne-university-seizure-prediction[master]git
$> datalad publish --since HEAD^            
2016-09-07 21:41:42,918 [INFO   ] Publishing dataset <Dataset path=/mnt/datasets/datalad/crawl/kaggle/melbourne-university-seizure-prediction> to sibling datalad-public ... (publish.py:270)
yoh@datasets.datalad.org's password: 
2016-09-07 21:41:45,734 [WARNING] Nothing was pushed. (publish.py:48)
> /home/yoh/proj/datalad/datalad/datalad/distribution/publish.py(283)__call__()
-> ds.repo.fetch(remote=dest_resolved)
(Pdb) n
> /home/yoh/proj/datalad/datalad/datalad/distribution/publish.py(284)__call__()
-> _log_push_info(ds.repo.push(remote=dest_resolved,
(Pdb) p dest_resolved
u'datalad-public'
(Pdb) l
279  	                import pdb; pdb.set_trace()
280  	                # Fetch first so we could investigate... actually it has a desired
281  	                # side-effect ATM -- merges git-annex of the remote so later push
282  	                # succeeds!
283  	                ds.repo.fetch(remote=dest_resolved)
284  ->	                _log_push_info(ds.repo.push(remote=dest_resolved,
285  	                                            refspec="+git-annex:git-annex"))
286  	
287  	            # we need to fetch
288  	            # TODO
289  	            # Note: This is about a gitpython issue as well as something about
(Pdb) n
> /home/yoh/proj/datalad/datalad/datalad/distribution/publish.py(285)__call__()
-> refspec="+git-annex:git-annex"))
(Pdb) 
2016-09-07 21:42:05,955 [WARNING] Nothing was pushed. (publish.py:48)
> /home/yoh/proj/datalad/datalad/datalad/distribution/publish.py(292)__call__()
-> ds.repo.fetch(remote=dest_resolved)
(Pdb) 
2016-09-07 21:42:37,620 [ERROR  ]  [bdb.py:dispatch_line:68] (BdbQuit) (main.py:252)
(venv-tests)3 12497 ->1.....................................:Wed 07 Sep 2016 09:42:37 PM EDT:.
(git)smaug:\u2026ts/datalad/crawl/kaggle/melbourne-university-seizure-prediction[master]git
$> git lg git-annex datalad-public/git-annex
* a2e2f31 - (datalad-public/git-annex) update (8 hours ago) [Yaroslav Halchenko]
* 36d1930 - branch created (8 hours ago) [Yaroslav Halchenko]
* 2a9872b - (git-annex) update (14 hours ago) [Yaroslav Halchenko]
* c569e24 - update (20 hours ago) [Yaroslav Halchenko]
* 39fba29 - update (20 hours ago) [Yaroslav Halchenko]
* 275aefb - update (20 hours ago) [Yaroslav Halchenko]
* b2166f3 - update (20 hours ago) [Yaroslav Halchenko]
* 02a6c8d - update (20 hours ago) [Yaroslav Halchenko]

```

so if you install `///kaggle/melbourne-university-seizure-prediction` you can't fetch any load since the git-annex branch on the webserver has no clue where the stuff is.

**Comment by yarikoptic** [comment]
for now did manually "git annex merge" (anyways we have only 1 git-annex remote) and then git push

so it feels that GitPython doesn't live to the premise of pushing `+git-annex` but the question is also why it happened for this particular repo and how could we guarantee that push of updated git-annex happens -- I guess we should at least check

**Comment by bpoldrack** [comment]
Did you try to reproduce the failure with an artificial repo?

**Comment by yarikoptic** [comment]
On September 8, 2016 7:46:19 AM EDT, Benjamin Poldrack <notifications@github.com> wrote:
>Did you try to reproduce the failure with an artificial repo? 
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/datalad/datalad/issues/765#issuecomment-245572702

I don't think so.. otherwise there would've been a cut paste it a PR

**Comment by bpoldrack** [comment]
Didn't manage to reproduce this yet. @yarikoptic : Still an issue? And if so, can you provide more info on how to get to that setup?

**Comment by yarikoptic** [comment]
oh well... as I said -- I thought that all those issues were addressed but then this one came about :-/
but I see the last commit on publish:
```
commit 50aba75abd135f09313d22b5dc89c872aa71d197
Author: Yaroslav Halchenko <debian@onerussian.com>
Date:   Thu Sep 22 11:27:54 2016 -0400

    BF: run annex merge before push, do not force the push of git-annex!
    
    Should Closes #835
```
which comes AFTER I filed this one... so I guess there is a good chance I did fix it up ;)  thus closing
